@xen-orchestra/rest-api/src/users/user.controller.mts

```diff
@@ -0,0 +1,65 @@
+import { Example, Get, Path, Query, Request, Response, Route, Security, Tags } from 'tsoa'
+import type { Request as ExRequest } from 'express'
+import { provide } from 'inversify-binding-decorators'
+import { notFoundResp, unauthorizedResp, type Unbrand } from '../open-api/common/response.common.mjs'
+import { XoController } from '../abstract-classes/xo-controller.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import type { XoUser } from '@vates/types'
+import { partialUsers, user, userIds } from '../open-api/oa-examples/user.oa-example.mjs'
+
+@Route('users')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('users')
+@provide(UserController)
+export class UserController extends XoController<XoUser> {
+  // --- abstract methods
+  async getAllCollectionObjects(): Promise<XoUser[]> {
+    const users = this.restApi.xoApp.getAllUsers()
+    return (await users).map(user => this.sanitizeUser(user))
+  }
+
+  async getCollectionObject(id: XoUser['id']): Promise<XoUser> {
+    const user = await this.restApi.xoApp.getUser(id)
+    return this.sanitizeUser(user)
+  }
+
+  private sanitizeUser(user: XoUser): XoUser {
```

Comment:
Do not use parameter properties like `private`. We are trying to maintain compatibility with `node 23.6` as much as possible so that we can run TypeScript natively with Node when XOA uses that version.
[See unsupported feature](https://devblogs.microsoft.com/typescript/announcing-typescript-5-8-beta/#the---erasablesyntaxonly-option)